GetJobber/atlantis
# chore: Clean up deps and npm audit issues

## Motivations

This PR cleans up a number of High Severity audit issues (from 19, down to 1). To achieve this, I tried to make the minimum amount of changes required. Along the way, I attempted to clean up some old, unnecessary dependencies as well. Check out my inline comments for more context.

This also closes a number of open Dependabot PRs:
* Closes https://github.com/GetJobber/atlantis/pull/2053
* Closes https://github.com/GetJobber/atlantis/pull/2026
* Closes https://github.com/GetJobber/atlantis/pull/2025
* Closes https://github.com/GetJobber/atlantis/pull/1996
* Closes https://github.com/GetJobber/atlantis/pull/1969
* Closes https://github.com/GetJobber/atlantis/pull/1959

Before |  After
:---:|:---:
<img width="577" alt="Screenshot 2024-10-08 at 11 39 35 AM" src="https://github.com/user-attachments/assets/e3f86152-a2ff-4772-aefd-1cff7d218e96"> | <img width="577" alt="Screenshot 2024-10-08 at 11 27 31 AM" src="https://github.com/user-attachments/assets/d682221a-e9eb-4d9e-b079-0fca5b1ce07f">


## Changes

### Changed

- Updated plop from v2 to v4
- Updated all storybook deps to match 7.6.20
- Updated react-native to match JM's version
    - Fixed some type-related issues due to this 

### Removed

- Unused dependencies

### Security

- Reduced 19 high sev issues down to 1

## Testing

To test this:

* install the pre-release versions ([comment](https://github.com/GetJobber/atlantis/pull/2058#issuecomment-2400507431)) inside JO/JF/JM
* poke around and make sure everything seems to look normal
    * I ran through creating client->request->quote->job->invoice as a basic flow to ensure things are working nicely
* I also ran through every storybook story to make sure they loaded fine. I didn't notice any problems there.
* Test out plop v4 via `npm run generate` to ensure it still works

Changes can be [tested via Pre-release](https://github.com/GetJobber/atlantis/actions/workflows/trigger-qa-build.yml)
